Dockerfile-alpine.template

```diff
@@ -1,15 +1,16 @@
 # Alpine Linux is not officially supported by the RabbitMQ team -- use at your own risk!
-FROM alpine:3.16
+FROM alpine:3.16 as build-base
 
 RUN apk add --no-cache \
-# grab su-exec for easy step-down from root
-		'su-exec>=0.2' \
-# bash for docker-entrypoint.sh
-		bash \
-# "ps" for "rabbitmqctl wait" (https://github.com/docker-library/rabbitmq/issues/162)
-		procps \
-# Bring in tzdata so users could set the timezones through the environment
-		tzdata
+	build-base \
+	dpkg-dev \
+	dpkg \
+	gnupg \
+	libc-dev \
+	linux-headers \
+	ncurses-dev
+
+FROM build-base as openssl-builder
```

Comment:
I'm not certain what `openssl-builder` being separate from `build-base` target really adds. It seems unlikely that we'd need to only build `build-base` while debugging. This applies to the Ubuntu Dockerfile as well.

We technically only need just a single target for the compiled artifacts, but it seems that having two would be slightly easier during Dockerfile development (for something like OpenSSL 3 changes).

Reply:
> it seems that having two would be slightly easier during Dockerfile development (for something like OpenSSL 3 changes)

That's the rationale.